autodist/graph_item.py

```diff
@@ -336,8 +365,48 @@ def var_op_name_to_grad_info(self):
                 if var_op.name in res:
                     raise ValueError('A variable cannot correspond to more than one update op for now.')
                 res[var_op.name] = expected_var_ops[var_op] + (op,)
+        self.updated = False
+        self.var_op_name_to_grad_dict = res
         return res
 
+    @property
+    def var_op_name_to_grad_info_optimize(self):
+        """An optimized version that is aware of this method is iteratively used."""
```

Comment:
please write a more thorough description of this method, explaining how this `v2` method improves over the previous one, in case in the future others need to fix anything here.